### DISCUSSION TITLE
Validating a value based on the properties of another object

### DISCUSSION THREAD

**Discussion by christophe-calmejane** [json-schema-org/community#581] [Q&A]
Hi,

I've been searching for this answer for 2 days, even asked chatGPT but nothing seems to work.

I have this json example file:
```json
{
  "uids": {
    "a" : {},
    "b" : {}
  },
  "uid": "c"
}
```

I need a schema that will validate that the string value for the `uid` property is one of the properties of `uids`.
In this example, it will fail since `c` isn't any property of `uids` (in this example, only `a` or `b` would be valid). This is of course a dynamic validation.
GPT offered me many things (including `$data`, `dependencies`) but nothing validated when testing at https://json-everything.net/json-schema/

Is it even possible (I don't mind using draft 2020-12), and if yes, how?
Thanks!

**Comment by gregsdennis** [comment]
Hi.  Unfortunately, matching against object _keys_ the way you want isn't possible.

There's a lot of ways to get _values_ using my [data vocab](https://docs.json-everything.net/schema/vocabs/data-2023/), but not keys.  The problem is that there isn't a good way to query for keys.

I think you're going to need to verify this in code.

**Comment by christophe-calmejane** [comment]
Hi thanks for the quick answer.
I guess I can change the json format to match this example: https://docs.json-everything.net/schema/vocabs/data-2023/#data-path-example
but I need to make sure `id` key is unique in the array.

Do you think it would be possible in the future (of json schema) to have access to the keys, so my example can be validated? I think it may help validate more models.

**Reply by gregsdennis** [reply]
It's possible, but as I mentioned it's a matter of finding a way to query for keys.

**Reply by gregsdennis** [reply]
> I need to make sure id key is unique in the array

I'd like to introduce you to my other vocab: https://docs.json-everything.net/schema/vocabs/array-ext/

**Reply by christophe-calmejane** [reply]
Oh ok I thought you said it was not possible _at all_ to validate my example.

Of course I find it better to use an object instead of array, since it inherently has unique keys, so there is no need to check values uniqueness (usng your array-ext, thanks btw).
Like you said there is the option to validate from the code (required anyway in my case since I need to access the other, omitted here, values for one specific uid).

I hope there will be some easy way (or a new vocab :)) to do this at some point.
